### DISCUSSION TITLE
Difficulty with custom fonts

### DISCUSSION THREAD

**Discussion by alancwoo** [dompdf/dompdf#2681] [Q&A] [answered]
I'm so sorry as I know you are being constantly bombarded with questions, especially about fonts, but I've tried almost every solution I could find to no avail and unfortunately it seems there are a lot of outdated posts around as well.

I I am trying to use a custom truetype font in a PDF:

```
<?php
require 'vendor/autoload.php';
use Dompdf\Dompdf;
use Dompdf\Options;

$options = new Options();
$options->set('fontDir', __DIR__ . '/fonts');
$options->set('tempDir', __DIR__ . '/tmp');
$options->set('fontCache', __DIR__ . '/tmp');\

$dompdf = new Dompdf($options);
$dompdf->setPaper('Tabloid');
$dompdf->loadHtml('
  <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
  <html>
  <head>
  <style>
    @font-face {
      font-family: "test";
      font-weight: normal;
      font-style: normal;
      src: url("fonts/test-regular.ttf") format("truetype");
    }

    body {
      font-family: "test", Helvetica, sans-serif;
    }
  </style>
  </head>
  <body>
    <div>Hello World</div>
  </body>');

$dompdf->render();
$dompdf->stream("dompdf_out.pdf", array("Attachment" => false));
```

My directory structure is like so:

```
/index.php
/fonts/test-regular.ttf
/fonts/test-regular-italic.ttf
```

I am using `dompdf ^1.1` and a simple php file hosted with [laravel valet](https://laravel.com/docs/8.x/valet). The ttf file is indeed accessible at `domain.test/fonts/test-regular.ttf`. If I use something like `Times`, I do see the corresponding afm files appearing in the `tmp` folder.

Many thanks for this library and hoping to find a solution!

**Comment by bsweeney** [comment, marked as answer]
By default, recent versions of Dompdf restrict local resources to the path specified by the chroot option. I believe all you need to do is set the chroot option to the directory where index.php is (or the ./fonts directory if you want to be more restrictive).

```
$options->set('chroot', __DIR__);
```

**Reply by alancwoo** [reply]
Thank you, this worked! I set the chroot in the options and all is working now:

```
$options = new Options();
$options->set('chroot', __DIR__);
$options->set('fontDir', __DIR__ . '/fonts');
$options->set('tempDir', __DIR__ . '/tmp');
$options->set('fontCache', __DIR__ . '/tmp');
...
```

Thank you so much again for this incredibly helpful library and for the help and the remarkably prompt response! Donation on the way!
